mypy/subtypes.py

```diff
@@ -249,13 +249,22 @@ def visit_overloaded(self, left: Overloaded) -> bool:
                     return True
             return False
         elif isinstance(right, Overloaded):
-            # TODO: this may be too restrictive
-            if len(left.items()) != len(right.items()):
+            if len(left.items()) < len(right.items()):
                 return False
-            for i in range(len(left.items())):
-                if not is_subtype(left.items()[i], right.items()[i], self.check_type_parameter,
+
+            # Ensure each overload in the left side is accounted for.
+            super_overloads = left.items()[:]
+            while super_overloads:
```

Comment:
Every overload in the supertype must have at least one corresponding overload in the subtype, yes. There are also some weird cases I suspect exist for each overload in the subtype and how it interacts with the supertype's overloads whose arguments it overlaps.  I gave an example in the test section below.